fix: Anvil widget borders

marks0351 opened the pull request:
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
## Description
 - fixed border design to match existing designs in fixed and auto.
 - Added crispness to canvas renders
 - fixed cropping of zone borders.
 - fixed issues with space distribution coz of overflow css

Bug: Zone highlights cropped
<img width="854" alt="Screenshot 2023-12-29 at 1 59 20 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/8523870e-a370-44e3-877b-1d0f402617b4">

Fixed: 
<img width="896" alt="Screenshot 2023-12-29 at 1 57 09 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/d575d6c1-13bb-414e-97c2-ed2642c0d03e">

Bug: Zone borders cropped
<img width="409" alt="Screenshot 2023-12-29 at 2 00 27 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/9881ef3f-d482-461c-af0e-6e6af2f6ed86">

Fixed: 
<img width="434" alt="Screenshot 2023-12-29 at 2 00 44 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/9b7db171-4746-4349-8194-1ba4cdf67306">

Bug: Borders in Dark mode
<img width="443" alt="Screenshot 2023-12-29 at 2 01 58 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/3a5e4578-0279-408b-9f6b-a59021ad94cb">

Fixed: 
<img width="437" alt="Screenshot 2023-12-29 at 2 01 38 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/d885e7f8-43ea-415d-b1a7-f49a0637d388">

Bug: Space distribution min width animation not happening

https://github.com/appsmithorg/appsmith/assets/35134347/2c8ea694-5f5f-4d15-a5a1-6ef3d35e795c

Fixed: 

https://github.com/appsmithorg/appsmith/assets/35134347/1192a2a8-f7d1-42a0-90ca-1ea58072e287

Box shadow looks consistent:
Before:
<img width="784" alt="Screenshot 2023-12-29 at 2 08 25 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/30e7fc46-a541-422c-ba8a-58603c9d43cb">
After:
<img width="838" alt="Screenshot 2023-12-29 at 2 09 32 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/93a98ae0-315a-4564-8f6b-7c3b0d42b5b5">

#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video
>
>
#### Type of change
> Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
## Testing
>
#### How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.
> Delete anything that is not relevant
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Test Plan
> Add Testsmith test cases links that relate to this PR
>
>
#### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)
>
>
>
## Checklist:
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag


#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new constant to manage widget outline offset.

- **Enhancements**
	- Improved visibility and styling of internal components with updated overflow handling.
	- Enhanced widget border styling by using shadow effects for better visual clarity.
	- Ensured consistent class naming in layout components with fail-safe defaults.

- **Bug Fixes**
	- Fixed canvas rendering issues by refining pixel ratio calculations and adjustments.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

marks0351 commented:
/build-deploy-preview skip-tests=true

riodeuno commented on app/client/src/ce/selectors/featureFlagsSelectors.ts:
```diff
@@ -10,6 +10,9 @@ export const selectFeatureFlagCheck = (
   flagName: FeatureFlag,
 ): boolean => {
   const flagValues = selectFeatureFlags(state);
+  if (flagName === "ab_wds_enabled") {
```
@marks0351 Please remove this.

marks0351 replied:
This is for the dp to work out will be removing it at last

riodeuno commented on app/client/src/layoutSystems/anvil/common/AnvilFlexComponent.tsx:
```diff
@@ -158,7 +158,7 @@ export function AnvilFlexComponent(props: AnvilFlexComponentProps) {
       position: "relative",
       // overflow is set to make sure widgets internal components/divs don't overflow this boundary causing scrolls
-      overflow: "visible",
+      overflow: "hidden",
```
Why did we revert this?

marks0351 replied:
this is not needed since content inside the last layer should not overflow.

marks0351 replied:
this change also makes the space redistribution min width shrinking buggy.

riodeuno replied:
This was added so that the zones and sections shadows can overflow the boundaries and not get chopped off.

riodeuno replied:
So how would we fix this then?

marks0351 replied:
AnvilFlexComponent is a common end layer for all widgets so enabling overflow here will not be the right thing to do. we could lose width and height internal to the section/zone to have box shadow showup without cropping instead of overflowing content. I see that sections have a bit of crop when there background is enabled zones look fine.

marks0351 replied:
https://github.com/appsmithorg/appsmith/issues/29975
Created an issue for this.

riodeuno commented on app/client/src/selectors/layoutSystemSelectors.ts:
```diff
@@ -5,6 +5,7 @@ import { LayoutSystemTypes } from "layoutSystems/types";
  * selector to fetch the application's layout type
  */
 export const getLayoutSystemType = (state: AppState) => {
+  return LayoutSystemTypes.ANVIL;
```
@marks0351 Please remove this.

riodeuno commented on app/client/src/utils/hooks/useFeatureFlag.ts:
```diff
@@ -4,6 +4,9 @@ import { selectFeatureFlags } from "@appsmith/selectors/featureFlagsSelectors";
 
 export function useFeatureFlag(flagName: FeatureFlag): boolean {
   const flagValues = useSelector(selectFeatureFlags);
+  if (flagName === "ab_wds_enabled") {
```
@marks0351 Please remove this.

marks0351 commented:
/build-deploy-preview skip-tests=true

riodeuno commented:
@marks0351 Please add a before and after screenshots to the PR description for each fix.

marks0351 commented:
Bug: Zone highlights cropped
<img width="854" alt="Screenshot 2023-12-29 at 1 59 20 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/8523870e-a370-44e3-877b-1d0f402617b4">

Fixed: 
<img width="896" alt="Screenshot 2023-12-29 at 1 57 09 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/d575d6c1-13bb-414e-97c2-ed2642c0d03e">

Bug: Zone borders cropped
<img width="409" alt="Screenshot 2023-12-29 at 2 00 27 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/9881ef3f-d482-461c-af0e-6e6af2f6ed86">

Fixed: 
<img width="434" alt="Screenshot 2023-12-29 at 2 00 44 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/9b7db171-4746-4349-8194-1ba4cdf67306">

Bug: Borders in Dark mode
<img width="443" alt="Screenshot 2023-12-29 at 2 01 58 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/3a5e4578-0279-408b-9f6b-a59021ad94cb">

Fixed: 
<img width="437" alt="Screenshot 2023-12-29 at 2 01 38 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/d885e7f8-43ea-415d-b1a7-f49a0637d388">

Bug: Space distribution min width animation not happening

https://github.com/appsmithorg/appsmith/assets/35134347/2c8ea694-5f5f-4d15-a5a1-6ef3d35e795c

Fixed: 

https://github.com/appsmithorg/appsmith/assets/35134347/1192a2a8-f7d1-42a0-90ca-1ea58072e287

Box shadow looks consistent:
Before:
<img width="784" alt="Screenshot 2023-12-29 at 2 08 25 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/30e7fc46-a541-422c-ba8a-58603c9d43cb">
After:
<img width="838" alt="Screenshot 2023-12-29 at 2 09 32 PM" src="https://github.com/appsmithorg/appsmith/assets/35134347/93a98ae0-315a-4564-8f6b-7c3b0d42b5b5">
@riodeuno

marks0351 commented:
/ok-to-test tags="All"
